Pod/Extensions/SSO/SKYWebOAuth.m

```diff
@@ -0,0 +1,200 @@
+//
+//  SKYWebOAuth.m
+//  SKYKit
+//
+//  Copyright 2015 Oursky Ltd.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#import "SKYWebOAuth.h"
+#import <SafariServices/SafariServices.h>
+
+@interface SKYWebOAuth () <SFSafariViewControllerDelegate>
+@end
+
+@implementation SKYWebOAuth {
+    BOOL _inProgress;
```

Comment:
Is it possible for the user to cancel the OAuth flow after initiated it? If the user cancelled the OAuth flow, is it possible for the user to re-try the OAuth flow?

Reply:
Yes, user can retry. `_inProgress` will be reset in `didCompleteOAuthFlow` and `didCompleteOAuthFlow` will be called when user cancel. (iOS9, 10 `SFSafariViewController ` delegate and iOS11 `SFAuthenticationSession` `completionHandler`)